src/main/webapp/app/entities/site-data/site-data-delete-dialog.controller.js

```diff
@@ -5,19 +5,31 @@
         .module('vivaxDataManagerApp')
         .controller('SiteDataDeleteController',SiteDataDeleteController);
 
-    SiteDataDeleteController.$inject = ['$uibModalInstance', 'entity', 'SiteData'];
+    SiteDataDeleteController.$inject = ['$uibModalInstance', 'entity', 'SiteData', 'ShareDataService'];
 
-    function SiteDataDeleteController($uibModalInstance, entity, SiteData) {
+    function SiteDataDeleteController($uibModalInstance, entity, SiteData, ShareDataService) {
         var vm = this;
         vm.siteData = entity;
+        var flag = ShareDataService.getFlag();
+        var siteDataId = ShareDataService.getSiteData().id;
         vm.clear = function() {
             $uibModalInstance.dismiss('cancel');
         };
-        vm.confirmDelete = function (id) {
-            SiteData.delete({id: id},
-                function () {
-                    $uibModalInstance.close(true);
-                });
-        };
+        if (!flag) {
```

Comment:
It would be less code to do:
var siteId; 
if(!flag){siteId = id)
else{siteId = siteDataId}

vn.confirmDeelete....